pkg/qdr/amqp_mgmt_mock_test.go

```diff
@@ -0,0 +1,159 @@
+package qdr
+
+import (
+	"flag"
+	"strings"
+	"testing"
+
+	"crypto/tls"
+	"gotest.tools/assert"
+)
+
+var clusterRun = flag.Bool("use-cluster", false, "run tests against a configured cluster")
+
+func TestQDR(t *testing.T) {
+
+	// AsInt -----------------------------------
+	goodAsInt := Record{}
+	goodAsInt["int8"] = int8(0)
+	goodAsInt["int16"] = int16(0)
+	goodAsInt["int32"] = int32(0)
+	goodAsInt["int64"] = int64(0)
+
+	goodAsInt["uint8"] = uint8(0)
+	goodAsInt["uint16"] = uint16(0)
+	goodAsInt["uint32"] = uint32(0)
+	goodAsInt["uint64"] = uint64(0)
+
+	for key, value := range goodAsInt {
+		_, ok := AsInt(value)
+		assert.Assert(t, ok, "AsInt failed on type |%s|.", key)
+	}
+
+	badAsInt := Record{}
+	badAsInt["float32"] = float32(0)
+	badAsInt["float64"] = float64(0)
+	badAsInt["string"] = "0"
+
+	for key, value := range badAsInt {
+		_, ok := AsInt(value)
+		assert.Assert(t, !ok, "AsInt succeeded on type |%s|, but should fail.", key)
+	}
+
+	// I guess we can at least test that these next few don't crash...
+	badAsInt.AsInt("string")
+	asConnection(badAsInt)
```

Comment:
you can build a particular record, and assert that asConnection returns a Connection object equal to what you expect ... 
This is just an example of comparing structs, not sure if it is the best way in go or there is another one ... 
https://github.com/skupperproject/skupper/blob/056b6020170a0cca12551cc3c8fded0fb5fe5bf7/cmd/skupper/skupper_test.go#L250